wasm/text/decoder.go

```diff
@@ -1,64 +1,741 @@
 package text
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/tetratelabs/wazero/wasm"
 )
 
+// parserPosition holds the positional state of a parser. Values are also useful as they allow you to do a reference
```

Comment:
this file used to be parser.go, but merging it with decoder.go caused enough changes it is basically a new file